### PR TITLE
🌱(project) add plugin boilerplate

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Add a complete working environment for the tdbp plugin.

## Proposal

- [x] add docker (compose) working environment
- [x] bootstrap frontend/api plugins

Note that CI/CD will be added in another PR.

### CODE REVIEW

File: src/api/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#
+# Warren Video package
```
Comment (quitterie-lcs) [inline src/api/pyproject.toml]:
```suggestion
# Warren package
```

File: src/api/warren_tdbp/api.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"""Warren API v1 tdbp router."""
+import logging
```
Comment (quitterie-lcs) [inline src/api/warren_tdbp/api.py:3]:
```suggestion

import logging
```

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+.venv
+env/
+venv/
+ENV/
+env.bak/
+venv.bak/
+
+# Project
+bin/patch_statements_date.py
+data
```
Comment (jmaupetit) [inline .gitignore]: To remove?

File: src/api/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#
+# Warren package
```
Comment (lebaudantoine) [inline src/api/pyproject.toml]: Warren tbdp ?

File: bin/git-hook-pre-commit [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+echo "pre-commit hook: checking trailing whitespace…"
+git diff-index --check --cached "$against" --
+
+# Run the necessary commands to check modified code sources.
+if git diff --staged --cached --name-only | grep -q 'src/api'; then
+  make lint-api
+fi
+if git diff --staged --cached --name-only | grep -q 'src/app'; then
+  make lint-app
+fi
```
Comment (lebaudantoine) [inline bin/git-hook-pre-commit]: to remove ?

**Review by lebaudantoine — COMMENTED** [review]
Niceee

**Review by quitterie-lcs — APPROVED** [review]
It all works well ! Let's go!

**Review by lebaudantoine — APPROVED** [review]

**Pull request merged**